x-pack/metricbeat/module/gcp/metrics/compute/metadata.go

```diff
@@ -6,17 +6,19 @@ package compute
 
 import (
 	"context"
+	"strconv"
 	"strings"
 	"time"
 
-	"github.com/pkg/errors"
-	"google.golang.org/api/compute/v1"
-	"google.golang.org/api/option"
-	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
-
+	compute "cloud.google.com/go/compute/apiv1"
```

Comment:
Why using this package instead of `google.golang.org/api/compute/v1`? I checked and it has [NewInstancesService](https://pkg.go.dev/google.golang.org/api/compute/v1#NewInstancesService) and [InstacesServiceAggregatedList](https://pkg.go.dev/google.golang.org/api/compute/v1#InstancesService.AggregatedList)

Reply:
According to [google docs](https://cloud.google.com/apis/docs/client-libraries-explained), [Cloud Client Libraries](https://cloud.google.com/apis/docs/cloud-client-libraries) are the recommended option. I made a proposal in issue #30098,  what do you think ?